src/main/java/com/oneops/api/util/TransitionUtil.java

```diff
@@ -0,0 +1,110 @@
+package com.oneops.api.util;
+
+import com.jayway.restassured.response.Response;
+import com.jayway.restassured.specification.RequestSpecification;
+import com.oneops.api.exception.OneOpsClientAPIException;
+import com.oneops.api.resource.model.CiResource;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class TransitionUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TransitionUtil.class);
+
+    public static void deployAndCheckStatus(String environmentName, String transitionEnvUri, RequestSpecification request,
+                                            JSONObject jsonObject) throws OneOpsClientAPIException {
+        // Start a new deployment
+        Response response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+        checkForErrors(environmentName, response);
+
+        // Wait for 5s before checking if the deployment started successfully
+        try {
+            Thread.sleep(10000);
+        } catch(InterruptedException ie) {
+            LOG.error("Error trying to delay: {}", ie.getMessage());
+        }
+
+        // Check for capacity issues and if deployment has started or failed due to capacity error.
+        response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+        checkForErrors(environmentName, response);
+    }
+
+    public static Response doDeploy(String environmentName, String transitionEnvUri,
+                                    RequestSpecification request, JSONObject jsonObject) throws OneOpsClientAPIException {
+
+        Response response = request.body(jsonObject.toString()).post(transitionEnvUri + environmentName + "/deployments");
+        if(response == null) {
+            String msg = String.format("Failed to start deployment for environment %s " +
+                    "due to null response" ,environmentName);
+            throw new OneOpsClientAPIException(msg);
+        }
+
+        return response;
+    }
+
+    public static void checkForErrors(String environmentName, Response response) throws OneOpsClientAPIException {
```

Comment:
1. what is this `checkforerrors` method doing exactly that wasn't required in the old api implementation? 
2. why are we parsing the error message and calling it a `comment`?

Reply:
I have updated the docstring the method with the reason for checking the comments of the API response. 
Adding it here for reference:
```
Errors during commit and deploy are part of comments of the Oneops API response, unless its a server error (status code 500).
So we need to parse the comments for errors.
As we need to call the deploy API twice
1st call - returns status if deployment plan is generated correctly
    a. If the comment contains an error that an active deployment is in progress,
    throw an exception
2nd call - returns one of the following -
    a. deployment plan generated successfully
    b. An error that an active deployment already in progress
    c. Error starting deployment for any reason
    (a) or (b) in comments indicate the deployment has started successfully
    (c) indicates there was a problem with starting the deployment.
    So, we should ignore the message is (a) or (b), but throw an exception if its (c)
```